src/main/host/managed_thread.rs

```diff
@@ -219,6 +219,18 @@ impl ManagedThread {
                             .unwrap();
                     }
 
+                    if !start_req.initial_working_dir_to_init.is_null() {
+                        // Write the working dir.
+                        let mut mem = ctx.process.memory_borrow_mut();
+                        let mut writer = mem.writer(ForeignArrayPtr::new(
+                            start_req.initial_working_dir_to_init,
+                            start_req.initial_working_dir_to_init_len,
+                        ));
+                        writer
+                            .write_all(ctx.process.current_working_dir().to_bytes_with_nul())
+                            .unwrap();
```

Comment:
Should there be a flush() call somewhere?

Reply:
Done